src/cascadia/TerminalControl/ControlCore.h

```diff
@@ -79,8 +79,9 @@ namespace winrt::Microsoft::Terminal::Control::implementation
 
         void SendInput(const winrt::hstring& wstr);
         void PasteText(const winrt::hstring& hstr);
-        bool CopySelectionToClipboard(bool singleLine, const Windows::Foundation::IReference<CopyFormat>& formats);
+        bool CopySelectionToClipboard(bool singleLine, const Windows::Foundation::IReference<CopyFormat>& formats, bool clearSelection = true);

```

Comment:
BTW I hope in the future that we can avoid such optional parameters. They usually lead to these "jack of all trades" / "world" functions. I mean what does selection clearing got to do with selection copying anyways?

In my view a better approach would be to just split that function up in those two parts. Then, if you actually need to clear the selection you simply call `ControlCore::ClearSelection()` yourself.

Reply:
Huh. Originally, I was hesitant to do that because TermControl shouldn't really have to clear the selection, I felt. But now that selection markers are in, I'm a lot more open to that idea. I'll pull this into a follow-up issue for the next release.

Tracking in #4993